# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/windows/sdn/sdnTextAccessible.h
# Commit: 1a948c0ec982
# Full Hash: 1a948c0ec9826d73bba860699f540cbb25f34e34
# Author: James Teh <jteh@mozilla.com>
# Date: 2021-04-29 21:42:31
# Regressor Bug: 1694865
# File Overlap Count: 1
# Description:
#   Bug 1694865 part 18: Make EnumVariant, GeckoCustom and sdn*Accessible classes aggregate MsaaAccessible/MsaaDocAccessible instead of AccessibleWrap/DocAccessibleWrap. r=morgan
#   
#   These objects need to aggregate the primary COM object (MsaaAccessible).
#   Once the IUnknown implementation moves out of AccessibleWrap, it won't be possible to aggregate *AccessibleWrap any more.
#   
# ==============================================================================

diff -r 3ea7b46bb4f8 -r 1a948c0ec982 accessible/windows/sdn/sdnTextAccessible.h
--- a/accessible/windows/sdn/sdnTextAccessible.h	Thu Apr 29 10:20:56 2021 +0000
+++ b/accessible/windows/sdn/sdnTextAccessible.h	Thu Apr 29 10:20:57 2021 +0000
@@ -10,7 +10,7 @@
 #include "ISimpleDOM.h"
 #include "IUnknownImpl.h"
 
-#include "AccessibleWrap.h"
+#include "MsaaAccessible.h"
 
 class nsIFrame;
 struct nsPoint;
@@ -20,8 +20,7 @@
 
 class sdnTextAccessible final : public ISimpleDOMText {
  public:
-  explicit sdnTextAccessible(AccessibleWrap* aAccessible)
-      : mAccessible(aAccessible){};
+  explicit sdnTextAccessible(MsaaAccessible* aMsaa) : mMsaa(aMsaa){};
   ~sdnTextAccessible() {}
 
   DECL_IUNKNOWN
@@ -61,7 +60,7 @@
   nsIFrame* GetPointFromOffset(nsIFrame* aContainingFrame, int32_t aOffset,
                                bool aPreferNext, nsPoint& aOutPoint);
 
-  RefPtr<AccessibleWrap> mAccessible;
+  RefPtr<MsaaAccessible> mMsaa;
 };
 
 }  // namespace a11y
